pkg/virt-controller/services/template.go

```diff
@@ -258,24 +260,23 @@ func (t *templateService) RenderLaunchManifestNoVm(vmi *v1.VirtualMachineInstanc
 	return t.renderLaunchManifest(vmi, nil, true)
 }
 
-func (t *templateService) RenderMigrationManifest(vmi *v1.VirtualMachineInstance, pod *k8sv1.Pod) (*k8sv1.Pod, error) {
-	imageIDs := containerdisk.ExtractImageIDsFromSourcePod(vmi, pod)
-	podManifest, err := t.renderLaunchManifest(vmi, imageIDs, false)
+func (t *templateService) RenderMigrationManifest(vmi *v1.VirtualMachineInstance, sourcePod *k8sv1.Pod) (*k8sv1.Pod, error) {
+	imageIDs := containerdisk.ExtractImageIDsFromSourcePod(vmi, sourcePod)
+	targetPod, err := t.renderLaunchManifest(vmi, imageIDs, false)
 	if err != nil {
 		return nil, err
 	}
 
-	if namescheme.PodHasOrdinalInterfaceName(network.NonDefaultMultusNetworksIndexedByIfaceName(pod)) {
-		ordinalNameScheme := namescheme.CreateOrdinalNetworkNameScheme(vmi.Spec.Networks)
-		multusNetworksAnnotation, err := multus.GenerateCNIAnnotationFromNameScheme(
-			vmi.Namespace, vmi.Spec.Domain.Devices.Interfaces, vmi.Spec.Networks, ordinalNameScheme, t.clusterConfig)
+	if t.netTargetAnnotationsGenerator != nil {
```

Comment:
@orelmisan Why do you need to check this? isn't it mandatory for this generator to be registered as part of `vca.templateService = services.NewTemplateService(...`?

Reply:
IIUC the migration controller is initialized with the template service instance, but the instance is already initialized and the generators were already registered via the template service init code:
```

	for _, opt := range opts {
		opt(&svc)
	}

```

Reply:
Thank you for reviewing this PR @enp0s3.

Since multiple unit tests use the concrete `templateService`, this condition is needed so these tests will not have to account for the network target annotations generator.

I think that our goal should be to eliminate this condition in the long run, which would require us to stub the templateService in multiple unit tests.

For completeness, the tests that use the concrete templateService are:
- pkg/storage/export/export/export_test.go
- pkg/storage/export/export/pvc-source_test.go
- pkg/storage/export/export/vm-source_test.go
- pkg/storage/export/export/vmsnapshot-source_test.go
- pkg/virt-controller/watch/application_test.go
- pkg/virt-controller/watch/migration_test.go
- pkg/virt-controller/watch/vmi_test.go

Reply:
We went with the same approch here for the same reason:
https://github.com/kubevirt/kubevirt/blob/097cc2a4538e21a628f69914da9ca2f629e891f2/pkg/virt-controller/services/template.go#L1459

Reply:
@enp0s3 , I agree the condition is ugly and very confusing.
There are only two options I can think of to overcome it:
- Make the hook-point a slice and then just iterate over. [This was not done because there are no other potential clients to this at this location]
- Initialize a noop function on construction before calling  the option initializers (i.e. defaults).

We could have the last option implemented here if you will support it. It will be checked implicitly by the existing tests that do not add the option explicitly.

Reply:
@EdDev Thank you for the clarification, 2nd option sounds good to me, anyway IMO we can do it in a follow up PR.